crates/holochain/src/core/workflow/publish_dht_ops_workflow/publish_query.rs

```diff
@@ -332,85 +413,143 @@ mod tests {
             ))
         };
 
-        db.write_async({
+        db.test_write({
             let query_state = state.clone();
 
-            move |txn| -> DatabaseResult<()> {
+            move |txn| {
                 let hash = query_state.as_hash().clone();
                 insert_op_authored(txn, &query_state).unwrap();
                 set_last_publish_time(txn, &hash, last_publish).unwrap();
                 set_receipts_complete(txn, &hash, facts.has_required_receipts).unwrap();
                 if facts.withold_publish {
                     set_withhold_publish(txn, &hash).unwrap();
                 }
-                Ok(())
             }
-        })
-        .await
-        .unwrap();
+        });
         state
     }
 
-    async fn test_data(db: &DbWrite<DbKindAuthored>, agent: AgentPubKey) -> Expected {
-        let mut results = Vec::new();
-        let cd = Consistent { this_agent: agent };
-        // We **do** expect any of these in the results:
-        // - Private: false.
-        // - WithinMinPeriod: false.
-        // - HasRequireReceipts: false.
-        // - IsThisAgent: true.
-        // - StoreEntry: true.
-        // - WitholdPublish: false.
-        let facts = Facts {
-            private: false,
-            within_min_period: false,
-            has_required_receipts: false,
-            is_this_agent: true,
-            store_entry: true,
-            withold_publish: false,
-        };
-        let op = create_and_insert_op(db, facts, &cd).await;
-        results.push(op);
-
-        // All facts are the same unless stated:
-
-        // - Private: true.
-        // - StoreEntry: false.
-        let mut f = facts;
```

Comment:
This and the following combination of facts is identical, but there weren't tests for all combination of `store_entry` and `private`.